packages/core/src/api/user.ts

```diff
@@ -0,0 +1,114 @@
+import { makeURLSearchParams, type REST } from '@discordjs/rest';
+import {
+	Routes,
+	type RESTGetAPICurrentUserGuildsQuery,
+	type RESTGetAPICurrentUserGuildsResult,
+	type RESTGetAPICurrentUserResult,
+	type RESTGetAPIUserResult,
+	type RESTGetCurrentUserGuildMemberResult,
+	type RESTPatchAPICurrentUserJSONBody,
+	type RESTPatchAPICurrentUserResult,
+	type RESTPatchAPIGuildMemberJSONBody,
+	type RESTPatchAPIGuildMemberResult,
+	type RESTPatchAPIGuildVoiceStateCurrentMemberJSONBody,
+	type RESTPatchAPIGuildVoiceStateCurrentMemberResult,
+	type RESTPostAPICurrentUserCreateDMChannelResult,
+	type Snowflake,
+} from 'discord-api-types/v10';
+
+export class UsersAPI {
+	public constructor(private readonly rest: REST) {}
+
+	public async get(userId: Snowflake) {
+		return this.rest.get(Routes.user(userId)) as Promise<RESTGetAPIUserResult>;
+	}
+
+	/**
+	 * Returns the user object of the requester's account
+	 */
+	public async getCurrentUser() {
+		return this.rest.get(Routes.user('@me')) as Promise<RESTGetAPICurrentUserResult>;
+	}
+
+	/**
+	 * Returns a list of partial guild objects the current user is a member of
+	 *
+	 * @param options - The options to use when fetching the current user's guilds
+	 */
+	public async getCurrentUserGuilds(options: RESTGetAPICurrentUserGuildsQuery = {}) {
+		return this.rest.get(Routes.userGuilds(), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
+		}) as Promise<RESTGetAPICurrentUserGuildsResult>;
+	}
+
+	/**
+	 * Leaves the guild with the given id
+	 *
+	 * @param guildId - The id of the guild
+	 */
+	public async leaveGuild(guildId: Snowflake) {
+		await this.rest.delete(Routes.userGuild(guildId));
+	}
+
+	/**
+	 * Edits the current user
+	 *
+	 * @param user - The new data for the current user
+	 */
+	public async editCurrentUser(user: RESTPatchAPICurrentUserJSONBody) {
```

Comment:
For this and for many other things, shouldn't the parameter be named `body` or `data`? We're not passing a `user` (but rather a partial of one).

Also, to avoid `api.user.editCurrentUser` (see duplicated `user`), can't we rename this to `update` or just `edit`?

Reply:
> For this and for many other things, shouldn't the parameter be named body or data? We're not passing a user (but rather a partial of one).

I've went ahead an done this, for areas where the json payload is more just optional options and/or query params I've kept them named as `options.`